consensus/consensus_tests/utils_test.go

```diff
@@ -304,17 +304,22 @@ func basePersistenceMock(t *testing.T, _ modules.EventsChannel) *modulesMock.Moc
 	ctrl := gomock.NewController(t)
 	persistenceMock := modulesMock.NewMockPersistenceModule(ctrl)
 	persistenceContextMock := modulesMock.NewMockPersistenceRWContext(ctrl)
+	persistenceReadContextMock := modulesMock.NewMockPersistenceReadContext(ctrl)
 
 	persistenceMock.EXPECT().Start().Return(nil).AnyTimes()
 	persistenceMock.EXPECT().SetBus(gomock.Any()).Return().AnyTimes()
-	persistenceMock.EXPECT().NewReadContext(int64(-1)).Return(persistenceContextMock, nil).AnyTimes()
+	persistenceMock.EXPECT().NewReadContext(gomock.Any()).Return(persistenceReadContextMock, nil).AnyTimes()
 	persistenceMock.EXPECT().ReleaseWriteContext().Return(nil).AnyTimes()
 
 	// The persistence context should usually be accessed via the utility module within the context
 	// of the consensus module. This one is only used when loading the initial consensus module
 	// state; hence the `-1` expectation in the call above.
```

Comment:
I am accessing the context for example here:

https://github.com/pokt-network/pocket/pull/374/files/dbb24d5918291dc28c8921bc2045de15180bc2e9#diff-9a2c4261d87cba56442982ea893473c0b02675123b3c6d09815ac15babef2cbdR64

and therefore we have Height that's != -1 as well. 

If I shouldn't be able to do that, we should probably look into ways for enforcing this "access constraint" in a more semantic way. 

I'll look into changing this so that we access the context via utility.

Stay tuned

Reply:
This comment now doesn't apply to this PR anymore but it does in #378. Referencing it there